File: _plugins/youtube.rb
class YouTube < Liquid::Tag
  Syntax = /^\s*([^\s]+)(\s+(\d+)\s+(\d+)\s*)?/

  def initialize(tagName, markup, tokens)
    super

    if markup =~ Syntax then
      @id = $1
    else
      raise "No YouTube ID provided in the \"youtube\" tag"
    end
  end

  def render(context)
    "<div class='video-container'><iframe src=\"https://www.youtube.com/embed/#{@id}\" frameborder=\"0\" allowfullscreen class='youtube-video'></iframe></div>"
  end

  Liquid::Template.register_tag "youtube", self
end

File: lib/components/YouTube.js
export default function YouTube({ token, vertical = false }) {
  return (
    <div className={`video-container ${vertical ? "vertical" : "horizontal"}`}>
      <iframe
        src={`https://www.youtube.com/embed/${token}`}
        title="YouTube video player"
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
        className="youtube-video"
      ></iframe>
    </div>
  );
}
